### PR TITLE
Word128 instance definitions

### PR DESCRIPTION
**Pull request by SamProtas**
I noticed this implementation of `Word128` is currently used by `cryptonite`. I'd like to port an existing pure haskell implementation of the Twofish block cipher from Hackage to `cryptonite`. These instance definitions would make this `Word128` implementation much easier to work with for that project.

### CODE REVIEW

**Comment by erikd** [comment]
If you want these instances *and* you want something more efficient than what is currently in `Data.Memory.ExtendedWords`, have a look at my [wide-word](https://hackage.haskell.org/package/wide-word) package.

**Comment by SamProtas** [comment]
@erikd thanks for the heads up. Your implementation clearly aims to squeeze out more performance than this PR.

TL;DR; this PR probabaly isn't necessary at all anymore. 

The motivation behind this PR (porting an existing Twofish implementation that uses Word128 to Cryptonite, a library trying to keep a small dependency footprint) has changed as I've discovered that implementation is broken.  

I've instead re-implemented it myself in Haskell based on the Golang implementation and have not made use of any 128bit data types.

Since I no longer plan on relying on these Word128 type classes and more efficient implementations exist elsewhere, this PR  doesn't seem to make much sense anymore.

**Comment by erikd** [comment]
@SamProtas No problem. I was just making you aware of the package.

**Comment by SamProtas** [comment]
There are more efficient implementations elsewhere.
